drivers/counter/counter_gd32_timer.c

```diff
@@ -0,0 +1,538 @@
+/*
+ * Copyright (c) 2017 - 2018, Nordic Semiconductor ASA
+ * Copyright (c) 2022 TOKITA Hiroshi <tokita.hiroshi@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gd_gd32_timer
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/gd32.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/irq.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/atomic.h>
+
+#include <gd32_timer.h>
+
+LOG_MODULE_REGISTER(counter_gd32_timer);
+
+#define TIMER_INT_CH(ch)  (TIMER_INT_CH0 << ch)
+#define TIMER_FLAG_CH(ch) (TIMER_FLAG_CH0 << ch)
+#define TIMER_INT_ALL	  (0xFFu)
+
+struct counter_gd32_ch_data {
+	counter_alarm_callback_t callback;
+	void *user_data;
+};
+
+struct counter_gd32_data {
+	counter_top_callback_t top_cb;
+	void *top_user_data;
+	uint32_t guard_period;
+	atomic_t cc_int_pending;
+	uint32_t freq;
+	struct counter_gd32_ch_data alarm[];
+};
+
+struct counter_gd32_config {
+	struct counter_config_info counter_info;
+	uint32_t reg;
+	uint16_t clkid;
+	struct reset_dt_spec reset;
+	uint16_t prescaler;
+	void (*irq_config)(const struct device *dev);
+	void (*set_irq_pending)();
```

Comment:
shouldn't there be `(void)`? Same as line below?

Reply:
Thanks for pointing out.
I fixed it.